modules/mono/glue/GodotSharp/GodotSharp/Core/Projection.cs

```diff
@@ -0,0 +1,817 @@
+
+            real_t x = 2 * near / (right - left);
+            real_t y = 2 * near / (top - bottom);
+
+            real_t a = (right + left) / (right - left);
+            real_t b = (top + bottom) / (top - bottom);
+            real_t c = -(far + near) / (far - near);
+            real_t d = -2 * far * near / (far - near);
+
+            return new Projection(
+                new Vector4(x, 0, 0, 0),
+                new Vector4(0, y, 0, 0),
+                new Vector4(a, b, c, -1),
+                new Vector4(0, 0, d, 0)
+            );
+        }
+
+        public static Projection CreateFrustumAspect(real_t size, real_t aspect, Vector2 offset, real_t near, real_t far, bool flipFov)
+        {
+            if (!flipFov)
+            {
+                size *= aspect;
+            }
+            return CreateFrustum(-size / 2 + offset.x, +size / 2 + offset.x, -size / aspect / 2 + offset.y, +size / aspect / 2 + offset.y, near, far);
+        }
+
+        public static Projection CreateLightAtlasRect(Rect2 rect)
+        {
+            return new Projection(
+                new Vector4(rect.Size.x, 0, 0, 0),
+                new Vector4(0, rect.Size.y, 0, 0),
+                new Vector4(0, 0, 1, 0),
+                new Vector4(rect.Position.x, rect.Position.y, 0, 1)
+            );
+        }
+
+        public static Projection CreateOrthogonal(real_t left, real_t right, real_t bottom, real_t top, real_t zNear, real_t zFar)
+        {
+            Projection proj = Projection.Identity;
+            proj.x.x = (real_t)2.0 / (right - left);
+            proj.w.x = -((right + left) / (right - left));
+            proj.y.y = (real_t)2.0 / (top - bottom);
+            proj.w.y = -((top + bottom) / (top - bottom));
+            proj.z.z = (real_t)(-2.0) / (zFar - zNear);
+            proj.w.z = -((zFar - zNear) / (zFar - zNear));
+            proj.w.w = (real_t)1.0;
+            return proj;
+        }
+
+        public static Projection CreateOrthogonalAspect(real_t size, real_t aspect, real_t zNear, real_t zFar, bool flipFov)
+        {
+            if (!flipFov)
+            {
+                size *= aspect;
+            }
+            return CreateOrthogonal(-size / 2, +size / 2, -size / aspect / 2, +size / aspect / 2, zNear, zFar);
+        }
+
+        public static Projection CreatePerspective(real_t fovyDegrees, real_t aspect, real_t zNear, real_t zFar, bool flipFov)
+        {
+            if (flipFov)
+            {
+                fovyDegrees = GetFovy(fovyDegrees, (real_t)1.0 / aspect);
+            }
+            real_t radians = Mathf.Deg2Rad(fovyDegrees / (real_t)2.0);
+            real_t deltaZ = zFar - zNear;
+            real_t sine = Mathf.Sin(radians);
+
+            if (deltaZ == 0) { GD.PushError("far is equal to near"); return Zero; }
+            if (sine == 0) { GD.PushError("sin of fov is zero"); return Zero; }
+            if (aspect == 0) { GD.PushError("aspect is zero"); return Zero; }
```

Comment:
Since the C++ equivalent doesn't push an error here, I don't think C# should either.

https://github.com/godotengine/godot/blob/8cb51a642e7c34a73e7ed877ab46c041521205e5/core/math/projection.cpp#L263-L265